packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts

```diff
@@ -499,6 +500,97 @@ export const createAzureCommunicationCallWithChatAdapter = async ({
   return new AzureCommunicationCallWithChatAdapter(callAdapter, chatAdapter);
 };
 
+/**
+ * A custom React hook to simplify the creation of {@link CallWithChatAdapter}.
+ *
+ * Similar to {@link createAzureCommunicationCallWithChatAdapter}, but takes care of asynchronous
+ * creation of the adapter internally.
+ *
+ * Allows arguments to be undefined so that you can respect the rule-of-hooks and pass in arguments
+ * as they are created. The adapter is only created when all arguments are defined.
+ *
+ * Note that you must memoize the arguments to avoid recreating adapter on each render.
+ * See storybook for typical usage examples.
+ *
+ * @beta
+ */
+export const useAzureCommunicationCallWithChatAdapter = (
+  /**
+   * Arguments to be passed to {@link createAzureCommunicationCallWithChatAdapter}.
+   *
+   * Allows arguments to be undefined so that you can respect the rule-of-hooks and pass in arguments
+   * as they are created. The adapter is only created when all arguments are defined.
+   */
+  args: Partial<AzureCommunicationCallWithChatAdapterArgs>,
+  /**
+   * Optional callback to modify the adapter once it is created.
+   *
+   * If set, must return the modified adapter.
+   */
+  afterCreate?: (adapter: CallWithChatAdapter) => Promise<CallWithChatAdapter>,
```

Comment:
The idea is to allow devs to monkey patch this adapter?
If yes, then it is no different from replacing object properties on the adapter resolved by this function.
This callback seems redundant.
What am I missing?

Reply:
This directly replaces some `useEffect` calls in our code.

The problem is: they have to monkey patch the adapter only once -- when it is created. So, by providing this argument, they get to do it in a single hook, instead of using another `useEffect` to observe when `adapter` changes.

Reply:
Right. I realised it when I looked at the implementation in the last file.